src/Bigtable/Connection/InstanceAdminConnectionInterface.php

```diff
@@ -0,0 +1,274 @@
+<?php
+/*
+ * Copyright 2018, Google LLC All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Bigtable;
+
+use Google\Cloud\Bigtable\ConnectionInterface;
+
+/**
+ * Connection to Bigtable InstanceAdmin API.
+ */
+interface InstanceAdminConnectionInterface extends ConnectionInterface
+{
+
+    /**
+     * Create an instance within a project.
+     *
+     * @param string   $instanceId   The ID to be used when referring to the new instance within its project,
+     *                               e.g., just `myinstance` rather than
+     *                               `projects/myproject/instances/myinstance`.
+     *
+     * @param string   $locationId   The unique location id.
+     *
+     * @param string   $clusterId    The unique id of the cluster to be create.
+     *
+     * @param array    $optionalArgs {
+     *                               Optional.
+     *
+     *     @type \Google\GAX\RetrySettings|array $retrySettings
+     *          Retry settings to use for this call. Can be a
+     *          {@see Google\GAX\RetrySettings} object, or an associative array
+     *          of retry settings parameters. See the documentation on
+     *          {@see Google\GAX\RetrySettings} for example usage.
+     * }
+     *
+     * @return TBD
+     *
+     */
+    public function createInstance($instanceId, $locationId, $clusterId, array $optionalArgs = []);
```

Comment:
[As demonstrated in other ConnectionInterface examples](https://github.com/GoogleCloudPlatform/google-cloud-php/blob/master/BigQuery/src/Connection/ConnectionInterface.php), the method should accept one argument, `array $args`, and will always return an array.

In this case, your implementation of this method should look like this:

```php
public function createInstance(array $args)
{
    return $this->send([$this->bigtableGapicClient, 'createInstance'], [
        $this->pluck('instanceId', $args),
        $this->pluck('locationId', $args),
        $this->pluck('clusterId', $args),
        $args
    ]);
}
```

Reply:
We need to be able to support all functionality that the create Instance proto allows.  That means that cluster needs to be an object, and we need a list of those.

Reply:
The "Connection" layer is an under the hood concept that acts as a bridge between the handwritten code and the upstream API. The reason we have this standard is so that it can be filled in with any transport implementation. This is not intended to be the code users interact with. As @jdpedrie mentioned [we have examples of this in practice](https://github.com/GoogleCloudPlatform/google-cloud-php/blob/master/Spanner/src/Connection/ConnectionInterface.php) already with libraries that are similar in scope to Bigtable.

Reply:
Is it ok for our object to be broken down into 3 interfaces?

We have three high level services:

- `InstanceAdmin`
- `TableAdmin`
- `Data`

@dwsupplee is it ok to have 3 different interfaces for each type of service?

Reply:
So is it ok to have one file with 38 RPCs across two different endpoints?

Reply:
That would be in line with what we did with Spanner for the connection layer, and will work.